From cb013b6602de32c647ed08faf899596664a18635 Mon Sep 17 00:00:00 2001
From: Jonathan Bell <jonathan@raspberrypi.com>
Date: Thu, 26 Oct 2023 13:47:54 +0100
Subject: [PATCH] arm: dt: bcm2712: don't unconditionally enable MPS read
 completions

RP1 supports it, but it's not a given that an arbitrary EP device
on PCIE2 will. Migrate the property to the rp1_target fragment.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2712-rpi-5-b.dts | 1 +
 arch/arm/boot/dts/bcm2712.dtsi        | 1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

--- a/arch/arm/boot/dts/bcm2712-rpi-5-b.dts
+++ b/arch/arm/boot/dts/bcm2712-rpi-5-b.dts
@@ -160,6 +160,7 @@
 };
 
 rp1_target: &pcie2 {
+	brcm,enable-mps-rcb;
 	brcm,vdm-qos-map = <0xbbaa9888>;
 	aspm-no-l0s;
 	status = "okay";
--- a/arch/arm/boot/dts/bcm2712.dtsi
+++ b/arch/arm/boot/dts/bcm2712.dtsi
@@ -1086,7 +1086,6 @@
 				      0x00 0x00000000
 				      0x10 0x00000000>;
 
-			brcm,enable-mps-rcb;
 			brcm,enable-l1ss;
 			status = "disabled";
 		};
